### PR TITLE
Final fixes for version 3

### PR DESCRIPTION
**Pull request by tux-rampage**
This PR contains updates to the documentation and a small incompatibility fix:

* Added a guide on how to use AoT in an Zend Expressive app
* Fixed ServiceManager parameter incompatibility in generated AoT factories

I also created an example repository for the app in this guide: https://github.com/lukanetconsult/zend-di-aot-example

### CODE REVIEW

File: docs/book/cookbook/aot-guide.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Using AoT In A ZendFramework App
```
Comment (froschdesign) [inline docs/book/cookbook/aot-guide.md]: Whitespace is missing: "Zend Framework"

File: docs/book/cookbook/aot-guide.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+You will learn how to:
+
+* Add a script to run the compilation
+* Use the generated injector with ServiceManager
+* Use the generated factories with ServiceManager
+
+## Create project and add zend-di
+
+For this guide will use an [expressive application](https://docs.zendframework.com/zend-expressive/)
```
Comment (froschdesign) [inline docs/book/cookbook/aot-guide.md]: Is this a "zend-expressive application" or "Zend Framework application"? I think there's a difference.

File: docs/book/cookbook/aot-guide.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+composer create-project zendframework/zend-expressive-skeleton zend-di-aot-example
+```
+
+Pick the components you want to use. As statet above, we will be using Zend ServiceManager
+and a Modular layout for this example.
+
+Now add zend-di with composer:
+
+```bash
+composer require zendframework/zend-di=^3.0
```
Comment (froschdesign) [inline docs/book/cookbook/aot-guide.md]: Why the version constraint?

File: docs/book/cookbook/aot-guide.md [hunk 4]
```diff
@@ -23,10 +23,11 @@ composer create-project zendframework/zend-expressive-skeleton zend-di-aot-examp
 Pick the components you want to use. As statet above, we will be using Zend ServiceManager
 and a Modular layout for this example.
 
-Now add zend-di with composer:
+Now add zend-di and zend-code with composer:
```
Comment (froschdesign) [inline docs/book/cookbook/aot-guide.md]: At this point it is not clear why zend-code is needed. Please add an explanation in the sentence above.

File: docs/book/cookbook/aot-guide.md (hunk 1 above)
Comment (froschdesign) [reply docs/book/cookbook/aot-guide.md]: See also my question below:

> Is this a "zend-expressive application" or "Zend Framework application"?

**Comment by tux-rampage** [comment]
@froschdesign Thanks for your feedback. I'll check the Details later.

File: docs/book/cookbook/aot-guide.md (hunk 2 above)
Comment (tux-rampage) [reply docs/book/cookbook/aot-guide.md]: I thought expressive was part of zend framework and an alternative to mvc. Anyway it's a good idea to change the title to make it more clear we're dealing with expressive here. I'll rephrase this paragraph accordingly.

File: docs/book/cookbook/aot-guide.md (hunk 3 above)
Comment (tux-rampage) [reply docs/book/cookbook/aot-guide.md]: I thought it states clearly that this requires 3.0 at minimum and prevents composer to accidentally install zend-di 2.x.
Should there be more explanation about this?

Comment (froschdesign) [reply docs/book/cookbook/aot-guide.md]: If version 3.0 is released, then version 3 will be installed automatically. The version constraint is not needed. Remove it and all is fine.
We should not over complicated the documentation.

Comment (weierophinney) [reply docs/book/cookbook/aot-guide.md]: I agree with both of you!

I'd suggest demonstrating _without_ the version constant, but adding a note that if v2 gets installed, you should try the above (with constraint) to determine what is blocking it. You might also want to consider adding some verbiage in that note that if you are upgrading from v2, you will need to remove zend-servicemanager-di first.

Comment (tux-rampage) [reply docs/book/cookbook/aot-guide.md]: Good points.
I totally missed the zend-servicemanager-di conflict. I'll add a Note and hint to the version constraint option as sugessted by @weierophinney

Comment (froschdesign) [reply docs/book/cookbook/aot-guide.md]: For version hint: 👍

**Comment by weierophinney** [comment]
Thanks, @tux-rampage! Edited for grammar and typography, but looks great!

**Pull request merged**